pcdsdevices/attenuator.py

```diff
@@ -203,6 +207,21 @@ def _setup_move(self, position):
             self._move_changed(value=moving_val)
             self._move_changed(value=self.done_value)
 
+    def subscribe(self, cb, event_type=None, run=True):
+        cid = super().subscribe(cb, event_type=event_type, run=run)
+        if event_type is None:
+            event_type = self._default_sub
+        if event_type == self.SUB_STATE and not self._has_subscribed_state:
+            for filt in self.filters:
+                filt.subscribe(self._run_filt_state, run=False)
```

Comment:
It's unintuitive, but it might be better to subscribe to the single "done" signal instead of to all the filter states. The "done" signal is the attenuator status readback. I'm imagining the callback being run up to 10 times after an attenuation change.

Reply:
Good point. Will modify